### PR TITLE
New Service: `s3tables`

### PR DESCRIPTION
**Pull request by gdavison**
### Description

Adds the following resource types for the service `s3tables`:

* `aws_s3tables_table_bucket`
* `aws_s3tables_table_bucket_policy`
* `aws_s3tables_table`
* `aws_s3tables_table_policy`
* `aws_s3tables_namespace`

### References

Closes https://github.com/hashicorp/terraform-provider-aws/issues/40407.

Announcement: https://aws.amazon.com/about-aws/whats-new/2024/12/amazon-s3-tables-apache-iceberg-tables-analytics-workloads/

Documentation: https://docs.aws.amazon.com/AmazonS3/latest/userguide/s3-tables.html

### Output from Acceptance Testing

```console
% make testacc PKG=s3tables TESTS=TestAccS3Tables

--- PASS: TestAccS3TablesTableBucket_disappears (24.10s)
--- PASS: TestAccS3TablesTableBucket_basic (24.44s)
--- PASS: TestAccS3TablesTable_disappears (28.60s)
--- PASS: TestAccS3TablesNamespace_basic (29.03s)
--- PASS: TestAccS3TablesTablePolicy_disappears (29.69s)
--- PASS: TestAccS3TablesTable_basic (32.72s)
--- PASS: TestAccS3TablesTablePolicy_basic (36.77s)
--- PASS: TestAccS3TablesTable_rename (44.76s)
--- PASS: TestAccS3TablesTableBucketPolicy_basic (23.78s)
--- PASS: TestAccS3TablesTable_updateNamespace (48.75s)
--- PASS: TestAccS3TablesTableBucketPolicy_disappears (20.27s)
--- PASS: TestAccS3TablesNamespace_disappears (19.76s)
--- PASS: TestAccS3TablesTableBucket_maintenanceConfiguration (59.77s)
--- PASS: TestAccS3TablesTable_maintenanceConfiguration (42.99s)
--- PASS: TestAccS3TablesTable_updateNameAndNamespace (38.87s)
```

### CODE REVIEW

File: website/docs/r/s3tables_namespace.html.markdown [hunk 1]
```diff
@@ -0,0 +1,62 @@
+---
+subcategory: "S3 Tables"
+layout: "aws"
+page_title: "AWS: aws_s3tables_namespace"
+description: |-
+  Terraform resource for managing an AWS S3 Tables Namespace.
```
Comment (ewbankkit) [inline website/docs/r/s3tables_namespace.html.markdown:6]: The branding is _Amazon S3 Tables_, not _AWS S3 Tables_.

**Comment by ewbankkit** [comment]
```console
% make testacc TESTARGS='-run=TestAccS3Tables' PKG=s3tables ACCTEST_PARALLELISM=3
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.23.3 test ./internal/service/s3tables/... -v -count 1 -parallel 3  -run=TestAccS3Tables -timeout 360m
go: downloading github.com/aws/aws-sdk-go-v2/service/s3tables v1.0.0
2024/12/03 16:31:27 Initializing Terraform AWS Provider...
=== RUN   TestAccS3TablesNamespace_basic
=== PAUSE TestAccS3TablesNamespace_basic
=== RUN   TestAccS3TablesNamespace_disappears
=== PAUSE TestAccS3TablesNamespace_disappears
=== RUN   TestAccS3TablesTableBucketPolicy_basic
=== PAUSE TestAccS3TablesTableBucketPolicy_basic
=== RUN   TestAccS3TablesTableBucketPolicy_disappears
=== PAUSE TestAccS3TablesTableBucketPolicy_disappears
=== RUN   TestAccS3TablesTableBucket_basic
=== PAUSE TestAccS3TablesTableBucket_basic
=== RUN   TestAccS3TablesTableBucket_disappears
=== PAUSE TestAccS3TablesTableBucket_disappears
=== RUN   TestAccS3TablesTableBucket_maintenanceConfiguration
=== PAUSE TestAccS3TablesTableBucket_maintenanceConfiguration
=== RUN   TestAccS3TablesTablePolicy_basic
=== PAUSE TestAccS3TablesTablePolicy_basic
=== RUN   TestAccS3TablesTablePolicy_disappears
=== PAUSE TestAccS3TablesTablePolicy_disappears
=== RUN   TestAccS3TablesTable_basic
=== PAUSE TestAccS3TablesTable_basic
=== RUN   TestAccS3TablesTable_disappears
=== PAUSE TestAccS3TablesTable_disappears
=== RUN   TestAccS3TablesTable_rename
=== PAUSE TestAccS3TablesTable_rename
=== RUN   TestAccS3TablesTable_updateNamespace
=== PAUSE TestAccS3TablesTable_updateNamespace
=== RUN   TestAccS3TablesTable_updateNameAndNamespace
=== PAUSE TestAccS3TablesTable_updateNameAndNamespace
=== RUN   TestAccS3TablesTable_maintenanceConfiguration
=== PAUSE TestAccS3TablesTable_maintenanceConfiguration
=== CONT  TestAccS3TablesNamespace_basic
=== CONT  TestAccS3TablesTablePolicy_disappears
=== CONT  TestAccS3TablesTable_updateNamespace
--- PASS: TestAccS3TablesNamespace_basic (17.06s)
=== CONT  TestAccS3TablesTable_maintenanceConfiguration
--- PASS: TestAccS3TablesTablePolicy_disappears (18.04s)
=== CONT  TestAccS3TablesTable_updateNameAndNamespace
--- PASS: TestAccS3TablesTable_updateNamespace (30.32s)
=== CONT  TestAccS3TablesTable_disappears
--- PASS: TestAccS3TablesTable_disappears (15.76s)
=== CONT  TestAccS3TablesTable_rename
--- PASS: TestAccS3TablesTable_maintenanceConfiguration (31.22s)
=== CONT  TestAccS3TablesTableBucket_basic
--- PASS: TestAccS3TablesTable_updateNameAndNamespace (33.10s)
=== CONT  TestAccS3TablesTablePolicy_basic
--- PASS: TestAccS3TablesTableBucket_basic (16.09s)
=== CONT  TestAccS3TablesTableBucket_maintenanceConfiguration
--- PASS: TestAccS3TablesTablePolicy_basic (19.29s)
=== CONT  TestAccS3TablesTableBucket_disappears
--- PASS: TestAccS3TablesTable_rename (29.23s)
=== CONT  TestAccS3TablesTable_basic
--- PASS: TestAccS3TablesTableBucket_disappears (13.17s)
=== CONT  TestAccS3TablesTableBucketPolicy_basic
--- PASS: TestAccS3TablesTable_basic (18.65s)
=== CONT  TestAccS3TablesTableBucketPolicy_disappears
--- PASS: TestAccS3TablesTableBucketPolicy_basic (16.30s)
=== CONT  TestAccS3TablesNamespace_disappears
--- PASS: TestAccS3TablesTableBucket_maintenanceConfiguration (40.46s)
--- PASS: TestAccS3TablesTableBucketPolicy_disappears (14.58s)
--- PASS: TestAccS3TablesNamespace_disappears (14.07s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/s3tables	119.417s
```

File: website/docs/r/s3tables_namespace.html.markdown (hunk 1 above)
Comment (gdavison) [reply website/docs/r/s3tables_namespace.html.markdown:6]: Good catch

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

**Pull request merged**